### PR TITLE
ARROW-12702: [JS] Update webpack and terser

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: js/gulp/minify-task.js [hunk 1]
```diff
@@ -62,13 +63,10 @@ const minifyTask = ((cache, commonConfig) => memoizeTask(cache, function minifyJ
             minimize: true,
             minimizer: [
                 new TerserPlugin({
-                    sourceMap: true,
                     terserOptions: {
                         ecma: terserLanguageNames[target],
-                        output: { comments: false, beautify: false },
-                        compress: { unsafe: true },
-                        mangle: true,
-                        safari10: true // <-- works around safari10 bugs, see the "safari10" option here: https://github.com/terser-js/terser#minify-options
```
Comment (domoritz) [inline js/gulp/minify-task.js:71]: Safari 10 is sufficiently old enough that we don't need this anymore.

File: js/gulp/minify-task.js [hunk 2]
```diff
@@ -83,7 +81,7 @@ const minifyTask = ((cache, commonConfig) => memoizeTask(cache, function minifyJ
 }))({}, {
     resolve: { mainFields: [`module`, `main`] },
     module: { rules: [{ test: /\.js$/, enforce: `pre`, use: [`source-map-loader`] }] },
-    output: { filename: '[name].js', library: mainExport, libraryTarget: `umd`, umdNamedDefine: true },
+    output: { filename: '[name].js', library: mainExport, libraryTarget: `umd`, umdNamedDefine: true, globalObject: 'this' },
```
Comment (domoritz) [inline js/gulp/minify-task.js:84]: No idea why this is needed.

**Review by trxcllnt — APPROVED** [review]

**Comment by kou** [comment]
@domoritz Could you rebase on master to merge this?

**Comment by domoritz** [comment]
I merged master so this should be good to squash and merge.

**Comment by kou** [comment]
Thanks. I'll merge this.

**Pull request closed without merging**